socks5.go

```diff
@@ -141,16 +154,19 @@ func (s *Server) ServeConn(conn net.Conn) error {
 	if err != nil {
 		err = fmt.Errorf("Failed to authenticate: %v", err)
 		s.config.Logger.Printf("[ERR] socks: %v", err)
+		promRequestFailed.Inc()
 		return err
 	}
 
 	request, err := NewRequest(bufConn)
 	if err != nil {
 		if err == unrecognizedAddrType {
```

Comment:
You should be able to just do one `promRequestFailed.Inc()` above this line and remove the following two